### ISSUE TITLE
Demo environment fails (websockets)

### ISSUE THREAD

**Issue by damnhipster** [elifesciences/elife-xpub#507]
This is due to WebSockets being requested over http when the page is https.

```
client.js:426 Mixed Content: The page at 'https://demo--xpub.elifesciences.org/' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws://demo--xpub.elifesciences.org/subscriptions'. This request has been blocked; this endpoint must be available over WSS.
SubscriptionClient.connect @ client.js:426
client.js:426 
```

**Comment by diversemix** [comment]
Potentially, we need to change the salt-stack used to deploy to the demo site. This is stored here:

https://github.com/elifesciences/elife-xpub-formula/blob/master/salt/elife-xpub/config/etc-nginx-sites-enabled-xpub.conf

This could contain the port forwarding as illustrated: http://nginx.org/en/docs/http/websocket.html

**Comment by diversemix** [comment]
The above suggested update to the config of nginx has been attempted - but does not work. @damnhipster will also try on the YLD infra, I will also try updating on the live `demo` site to see what the issue is. 

This has highlighted another issue - one of the container size being so large that we cannot do a test deployment locally in a VM - as the network keeps dropping. This also might be due to the office environment too. There is a fix for this in `245-prod` to dramatically reduce the size of the container. Going to raise a separate issue for this.

**Comment by diversemix** [comment]
New issue #517

**Comment by damnhipster** [comment]
Websocket requests can be made using wss:// rather than ws:// on a secure page. The demo and staging environments on Kubernetes support this out of the box. What still needs to be done is changing the client to use wss:// rather than ws:// if the page is requested over https. This logic sits within pubsweet: https://gitlab.coko.foundation/pubsweet/pubsweet/merge_requests/370

**Comment by damnhipster** [comment]
I've raised #518 to roll back the changes, as it'll require a change in pubsweet-client to support `wss://`.

**Comment by diversemix** [comment]
Reverting is a good idea in this instance. I'll mark this as blocked ... while you get an MR for those changes (https://gitlab.coko.foundation/pubsweet/pubsweet/commit/16ab81c940a5cce58d1988499c63c0541786e47e ) ... and we get a new release :+1:

**Comment by damnhipster** [comment]
Waiting on https://gitlab.coko.foundation/pubsweet/pubsweet/merge_requests/380 to be released. This should allow us to bring back the changes introduced in #368

**Comment by diversemix** [comment]
We now have `pubsweet-client@4.1.2` released
